spec/std/big/big_decimal_spec.cr

```diff
@@ -182,6 +182,21 @@ describe BigDecimal do
     1.5.to_big_f.to_big_d.should eq (BigDecimal.new(15, 1))
   end
 
+  it "can be converted from scientific notation" do
+    "-0.123e12".to_big_d.should eq (BigDecimal.new("-123000000000"))
+    "0.123e12".to_big_d.should eq (BigDecimal.new("123000000000"))
+    "0.123e+12".to_big_d.should eq (BigDecimal.new("123000000000"))
+    "-0.123e-7".to_big_d.should eq (BigDecimal.new("-0.0000000123"))
+    "-0.1e-7".to_big_d.should eq (BigDecimal.new("-0.00000001"))
+    "0.1e-7".to_big_d.should eq (BigDecimal.new("0.00000001"))
+    "1.0e-8".to_big_d.should eq (BigDecimal.new("0.00000001"))
+    "10e-8".to_big_d.should eq (BigDecimal.new("0.0000001"))
+    "1.0e+8".to_big_d.should eq (BigDecimal.new("100000000"))
+    "10e+8".to_big_d.should eq (BigDecimal.new("1000000000"))
+    "10E+8".to_big_d.should eq (BigDecimal.new("1000000000"))
+    "10E8".to_big_d.should eq (BigDecimal.new("1000000000"))
```

Comment:
What about tests for `10.01e1 == 100.1` and `10.01e-1 == 1.001`? Not all exponents are large

Reply:
Good catch! 🎉 Former case (`10.01e1`) uncovered bug in the implementation for which fix is included in the following commit.